apps/api/src/server/applications/application/documents/document.controller.js

```diff
@@ -82,3 +82,28 @@ export const updateDocuments = async ({ body }, response) => {
 	}
 	response.send(items);
 };
+
+/**
+ * Gets the blob storage uri components for a single document
+ * blobStorageContainer and blobStoragePath
+ *
+ * @type {import('express').RequestHandler<{guid: string}, ?, ?, any>}
+ */
+export const getDocumentUri = async ({ params }, response) => {
+	let /** @type { import('apps/api/prisma/schema.js').Document |null} */ document = null;
+
+	try {
+		document = await documentRepository.getById(params.guid);
+
+		if (document === null || typeof document === 'undefined') {
+			throw new Error(`Unknown document guid ${params.guid}`);
+		}
```

Comment:
This looks like it should live in a validator function in the `routes`?

Reply:
Yes I was initially going to have a validation fn, but it would have to make the same getByID call as in this fn, and I saw little value in calling the same fn twice separately

Reply:
I wonder if we can do some operations in the validator and then save the data (such as the case obtained from the DB) and then send it over to the controller

Reply:
this can be a bit difficult to handle, but on the antivirus-check branch there is a validator named "validateDocumentGUID", which I believe does exactly what is happening here. 
And yes, apparently that one uses the getById fn so maybe it could be used/optimized?

Reply:
after discussion with Mayya, agreed to keep code as is for now, but have a dev chat about a consistent way forward for similar stuff in the future